src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -9,18 +9,148 @@
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private Node<K, V>[] nodeArray;
+    private int size;
+    private final static int DEFAULT_CAPACITY = 16;
+    private final static float LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private int newArrayIndex;
+
+    public MyHashMap() {
+        this.nodeArray = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        this.threshold = thresholdCalculation(nodeArray.length);
+    }
+
     @Override
     public void put(K key, V value) {
-        
+        int index = findIndex(key);
+        if (threshold == size) {
+            reSize();
+        }
+        addNewNode(key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        newArrayIndex = findIndex(key);
+        Node<K, V> nodeWithNewIndex = nodeArray[newArrayIndex];
+        if (Objects.isNull(nodeWithNewIndex)) {
+            return null;
+        }
+        if (Objects.equals(nodeWithNewIndex.key, key)) {
+            return nodeWithNewIndex.value;
+        }
+        if (Objects.nonNull(nodeWithNewIndex.next)) {
+            Node<K, V> nextKey = nodeWithNewIndex.next;
+            while (nextKey != null) {
+                if (Objects.equals(nextKey.key, key)) {
+                    return nextKey.value;
+                }
+                nextKey = nextKey.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private boolean isEmpty(int arrayIndex) {
+        return Objects.isNull(nodeArray[arrayIndex]);
+    }
+
+    private void reSize() {
+        int newArraySize = nodeArray.length * 2;
+        Node<K, V>[] newNodeArray = (Node<K, V>[]) new Node[newArraySize];
+        thresholdCalculation(newArraySize);
+        size = 0;
+        Node<K, V>[] oldNodeArray = nodeArray;
+        nodeArray = newNodeArray;
+        for (Node<K, V> oldNode : oldNodeArray) {
+            if (Objects.nonNull(oldNode)) {
+                newArrayIndex = findIndex(oldNode.key);
+                addNewNode(oldNode.key, oldNode.value, newArrayIndex);
+                Node<K, V> nextNode = oldNode.next;
+                while (nextNode != null) {
+                    int newNodeIndex = findIndex(nextNode.key);
+                    addNewNode(nextNode.key, nextNode.value, newNodeIndex);
+                    nextNode = nextNode.next;
+                }
+            }
+        }
+    }
+
+    private int findIndex(K key) {
+        return Objects.isNull(key) ? 0 : Math.abs(hash(key) % nodeArray.length);
+    }
+
+    private void addNewNode(K key, V value, int arrayIndex) {
+        Node<K, V> nodeByIndex = nodeArray[arrayIndex];
+        if (isEmpty(arrayIndex)) {
```

Comment:
just check node you get before 'if' from array by index (isNull(nodeByIndex)). Also remove isEmpty() method

Reply:
@al-pavlenko +